app/services/teacher_training_public_api/sync_courses.rb

```diff
@@ -41,6 +41,8 @@ def create_or_update_course(course_from_api, recruitment_cycle_year, incremental
       add_accredited_provider(course, course_from_api[:accredited_body_code], recruitment_cycle_year)
 
       new_course = course.new_record?
+      course_in_next_recruitment_cycle = course.recruitment_cycle_year == 2022
```

Comment:
Can this check either be moved inline inline (on line 54) or extracted to a method as it's hard to read as is, because of the assignment and check.